connections/oikotie/mapper.py

```diff
@@ -0,0 +1,358 @@
+
+
+def map_apartment_type(elastic_apartment: ElasticApartment) -> ApartmentType:
+    project_building_type = elastic_apartment.project_building_type
+    if project_building_type == "Kerrostalo":
+        return ApartmentType.BLOCK_OF_FLATS
+    elif project_building_type == "Rivitalo":
+        return ApartmentType.ROW_HOUSE
+    elif project_building_type == "Omakotitalo":
+        return ApartmentType.HOUSE
+    else:
+        raise ValueError(
+            _("could not map the building type %s") % project_building_type
+        )
+
+
+def map_mode_of_habitation(elastic_apartment: ElasticApartment) -> ModeOfHabitation:
+    if elastic_apartment.project_holding_type == "Asumisoikeushuoneisto":
+        mode_of_habitation_type = ModeOfHabitationType.RIGHT_OF_OCCUPANCY
+    elif elastic_apartment.project_holding_type == "Osakehuoneisto":
+        mode_of_habitation_type = ModeOfHabitationType.OWNED
+    else:
+        raise ValueError(
+            _("could not map the mode of habitation %s")
+            % elastic_apartment.project_holding_type
+        )
+    return ModeOfHabitation(type=mode_of_habitation_type)
+
+
+def map_city(elastic_apartment: ElasticApartment) -> City:
+    return City(id=1, value=elastic_apartment.project_city)
+
+
+def get_estate() -> Estate:
+    return Estate(type=EstateType.REAL_ESTATE)
+
+
+def map_pictures(elastic_apartment: ElasticApartment) -> List[Picture]:
+    pictures = [
+        Picture(
+            index=1, is_floor_plan=False, url=elastic_apartment.project_main_image_url
+        )
+    ]
+    for idx, picture_url in enumerate(elastic_apartment.project_image_urls):
+        pictures.append(Picture(index=idx + 2, is_floor_plan=False, url=picture_url))
+    return pictures
+
+
+def map_floor_location(elastic_apartment: ElasticApartment) -> FloorLocation:
+    high = elastic_apartment.floor == elastic_apartment.floor_max
+    low = elastic_apartment.floor == 1
+    return FloorLocation(
+        high=high,
+        low=low,
+        number=elastic_apartment.floor,
+        count=elastic_apartment.floor_max,
+        description="",
+    )
+
+
+def map_balcony(elastic_apartment: ElasticApartment) -> Balcony:
+    return Balcony(
+        value=elastic_apartment.has_balcony,
+        description=elastic_apartment.balcony_description,
+    )
+
+
+def map_living_area(elastic_apartment: ElasticApartment) -> LivingArea:
+    return LivingArea(unit="m2", area=elastic_apartment.living_area)
+
+
+def map_lift(elastic_apartment: ElasticApartment) -> Lift:
+    return Lift(value=elastic_apartment.project_has_elevator, description="")
+
+
+def map_year_of_building(elastic_apartment: ElasticApartment) -> YearOfBuilding:
+    return YearOfBuilding(
+        original=elastic_apartment.project_construction_year, description=""
+    )
+
+
+def get_general_condition() -> GeneralCondition:
+    return GeneralCondition(level=GeneralConditionLevel.NEW, description="")
+
+
+def get_site() -> Site:
```

Comment:
Kind of same here?

Reply:
Yea, the apartments are always owned, not rented so I left it as constant.